upb/pb/decoder.c

```diff
@@ -917,27 +965,6 @@ void upb_pbdecoder_reset(upb_pbdecoder *d) {
   d->residual_end = d->residual;
```

Comment:
So I reset skip in start(), which is in line with where I reset other things of its type. I haven't quite figured out yet what should be reset in reset() vs start() yet. I don't know what the semantics should be around: should a decoder be able to be used multiple times without a reset()? Should reset() just set some bit that allows for a start() when you weren't already in an initial state? Still figuring all this out. But the current code should work.